crates/astria-conductor/src/block_verifier.rs

```diff
@@ -153,18 +149,21 @@ impl BlockVerifier {
             .wrap_err("failed to get validator set")?;
 
         // find proposer address for this height
-        let expected_proposer_address = validator_set
-            .get_proposer()
-            .wrap_err("failed to get proposer from validator set")?
-            .address;
+        let expected_proposer_address = public_key_bytes_to_address(
+            validator_set
+                .get_proposer()
+                .wrap_err("failed to get proposer from validator set")?
+                .pub_key
+                .key
+                .0
```

Comment:
I'm not usually too concerned with this, but this violates the law of demeter too badly.

Can you provide a function that takes the pub key and generates the final result from there?